metadata/db.go

```diff
@@ -0,0 +1,180 @@
+package metadata
+
+import (
+	"context"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/boltdb/bolt"
+	"github.com/containerd/containerd/gc"
+	"github.com/containerd/containerd/log"
+	"github.com/pkg/errors"
+)
+
+type DB struct {
+	db *bolt.DB
+
+	rlock sync.RWMutex
+	mlock sync.RWMutex
```

Comment:
Is this m for "modification"? Would `wlock` be better?

Reply:
Short for "mutation", modification works as well. `wlock` would probably be clearer

Reply:
Alright, wlock sounds like fine. Just make sure to make it clear how these work together as a latch.